models/users.js

```diff
@@ -117,15 +117,23 @@ const addOrUpdate = async (userData, userId = null) => {
     }
 
     // userId is null, Add or Update user
-    let user;
+    let user = null;
+
     if (userData.github_user_id) {
       user = await userModel.where("github_user_id", "==", userData.github_user_id).limit(1).get();
     }
-    if (!user || (user && user.empty)) {
+
+    if (userData.github_id && (!user || (user && user.empty))) {
       user = await userModel.where("github_id", "==", userData.github_id).limit(1).get();
     }
+
+    if (userData.email && (!user || (user && user.empty))) {
```

Comment:
- if we're already doing a `!user` check do we need to do `(user && user.empty))` can we just do `!user || user.empty` ?

Reply:
I'll update the condition to !user || user.empty to simplify it.